arcgis-ios-sdk-samples/Content Display Logic/Controllers/ContentCollectionViewController.swift

```diff
@@ -14,39 +14,21 @@
 
 import UIKit
 
-class CustomFlowLayout:UICollectionViewFlowLayout {
-    
-    override func shouldInvalidateLayout(forBoundsChange newBounds: CGRect) -> Bool {
-        return true
-    }
-}
-
-private let reuseIdentifier = "CategoryCell"
-
-class ContentCollectionViewController: UIViewController, UICollectionViewDataSource, UICollectionViewDelegate, UICollectionViewDelegateFlowLayout {
+class ContentCollectionViewController: UICollectionViewController, UICollectionViewDelegateFlowLayout {
 
-    @IBOutlet private var collectionView:UICollectionView!
     @IBOutlet private var collectionViewFlowLayout:UICollectionViewFlowLayout!
     
     /// The categories to display in the collection view.
-    var categories:[Category] = []
-
-    private var transitionSize:CGSize!
+    var categories:[Category] = []{
+        didSet{
+            // add search only after setting categories to ensure that the samples are available
+            addSearchController()
+        }
+    }
     
     // strong reference needed for iOS 10
     var searchController:UISearchController?
     
-    override func viewDidLoad() {
-        super.viewDidLoad()
-        
-        if #available(iOS 11.0, *) {
-            collectionView.contentInsetAdjustmentBehavior = .never
-        } else {
-            automaticallyAdjustsScrollViewInsets = false
-        }
-        
-        addSearchController()
-    }
     override func viewWillAppear(_ animated: Bool) {
         super.viewWillAppear(animated)
         // ensure that the search results appear beneath the navigation bar
```

Comment:
Is modifying `definesPresentationContext` in `viewWillAppear(_:)` and `viewWillDisappear(_:)` still needed after making these changes?

Reply:
Unfortunately yes, or else the search field cannot be selected on iOS 10. I'll add `#available(iOS 11.0, *)` checks to make it a bit clearer.